src/Search/Queue/WatchableQueue.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+namespace eLife\Search\Queue;
+
+interface WatchableQueue
+{
+    /**
+     * Adds item to the queue.
+     *
+     * Mock: Add item to queue.
+     * SQS: This will set the queue item into the memory slot for re-processing.
+     */
+    public function enqueue(QueueItem $item) : bool;
+
+    /**
+     * Starts process of removing item.
+     *
+     * Mock: Move to separate "in progress" queue.
+     * SQS: this will change the timeout of the in-memory item.
+     */
+    public function dequeue(int $timeout = 30) : QueueItem;
+
+    /**
+     * Commits to removing item from queue, marks item as done and processed.
+     *
+     * Mock: Remove item completely.
+     * SQS: this will delete the item from the queue.
+     */
```

Comment:
I do think that hooking Gearman up to MySQL would simplify this quite a lot, make it safer anyway. That would allow Gearman to handle its own recovery and re-trying, like it does currently with the batch processing.

Is the MySQL integration with Gearman transparent in terms of the implementation, or would we have to change some specifics?